### ISSUE TITLE
fr: deprecate MeshConfig and proxyConfig methods for configuring telemetry

### ISSUE THREAD

**Issue by craigbox** [istio/istio#54003]
per @zirain now that Telemetry API is v1, we should start the process of deprecating the `meshConfig` and `proxy.istio.io/config` annotation methods of configuring telemetry.

**Comment by zirain** [comment]
I don't `proxy.istio.io/config` is used for telemetry only.

**Comment by craigbox** [comment]
Yeah, nor is meshconfig. Talking about the relevant legacy telemetry config in both cases.

**Comment by zirain** [comment]
xref: https://github.com/istio/istio/issues/37751

**Comment by craigbox** [comment]
I think I take from that:

a. MeshConfig.ExtensionProvider.EnvoyOpenTelemetryLogProvider.LogFormat is equivalent to MeshConfig.accessLogFormat
b. The behavior is the same, so no reason to block deprecation
c. There's a desire to be able to configure that in the Telemetry API instead

?

**Comment by zirain** [comment]
a. it's simple to use built-in provider(named envoy)
b. yes
c. yes

**Comment by zirain** [comment]
not stale
